### PR TITLE
Add update.js

### PR DESCRIPTION
**Pull request by ljhaywar**

### CODE REVIEW

File: update.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+ * Update an Airbnb listing with the given name
+ * Note: If more than one listing has the same name, only the first listing the database finds will be updated.
+ *       It's best to use updateOne when querying on fields that are guaranteed to be unique.
+ * @param {MongoClient} client A MongoClient that is connected to a cluster with the sample_airbnb database
+ * @param {string} nameOfListing The name of the listing you want to update
+ * @param {object} updatedListing An object containing all of the properties to be updated for the given listing
+ */
+async function updateListingByName(client, nameOfListing, updatedListing) {
+    // See http://bit.ly/Node_updateOne for the updateOne() docs
+    result = await client.db("sample_airbnb").collection("listingsAndReviews").updateOne({ name: nameOfListing }, { $set: updatedListing });
```
Comment (daprahamian) [inline update.js]: `const result =`

File: update.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+ *       It's best to use updateOne when querying on fields that are guaranteed to be unique.
+ * Note: For educational purposes, we have split the update and upsert functionality into separate functions.
+ *       Another option is to have a single function where a boolean param indicates if the update should be an upsert. 
+ * @param {MongoClient} client A MongoClient that is connected to a cluster with the sample_airbnb database
+ * @param {string} nameOfListing The name of the listing you want to upsert
+ * @param {object} updatedListing An object containing all of the properties to be upserted for the given listing
+ */
+async function upsertListingByName(client, nameOfListing, updatedListing) {
+    // See http://bit.ly/Node_updateOne for the updateOne() docs
+    result = await client.db("sample_airbnb").collection("listingsAndReviews").updateOne({ name: nameOfListing }, { $set: updatedListing }, { upsert: true });
```
Comment (daprahamian) [inline update.js]: `const result =`

File: update.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    }
+}
+
+/**
+ * Update all Airbnb listings that do not have a property type so they have property_type 'Unknown'
+ * @param {MongoClient} client A MongoClient that is connected to a cluster with the sample_airbnb database
+ */
+async function updateAllListingsToHavePropertyType(client) {
+    // See http://bit.ly/Node_updateMany for the updateMany() docs
+    result = await client.db("sample_airbnb").collection("listingsAndReviews").updateMany({ property_type: { $exists: false } }, { $set: { property_type: "Unknown" } });
```
Comment (daprahamian) [inline update.js]: `const result =`

File: update.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+/**
+ * Print an Airbnb listing with the given name
+ * Note: If more than one listing has the same name, only the first listing the database finds will be printed.
+ *       It's best to use findOne when querying on fields that are guaranteed to be unique.
+ * @param {MongoClient} client A MongoClient that is connected to a cluster with the sample_airbnb database
+ * @param {String} nameOfListing The name of the listing you want to find
+ */
+async function findListingByName(client, nameOfListing) {
+    // See http://bit.ly/Node_findOne for the findOne() docs
+    result = await client.db("sample_airbnb").collection("listingsAndReviews").findOne({ name: nameOfListing });
```
Comment (daprahamian) [inline update.js]: `const result = `

File: update.js (hunk 1 above)
Comment (ljhaywar) [reply update.js]: You've called me on this before.  Thanks for catching them!  I'll try to catch them in the future.

**Pull request merged**